### ISSUE TITLE
sparta binaries fail to execute when not built from primary directory

### ISSUE THREAD

**Issue by kynrai** [mweagle/Sparta#29]
when calling go run main.go provision with go1.7.3, the lambda functions will not work when invoked, instead the binary itself crashes with exit code 2.

repeating the process with go1.6.2 will work.

**Comment by mweagle** [comment]
Hi @kynrai,
I'm not able to reproduce this locally using https://github.com/mweagle/SpartaHelloWorld and 1.7.3. Can you try deploying with `--level debug` and if the problem persists, post the logfile as a gist?

Do you see the same problem if you run locally with `go run main.go --level info explore`? Thanks.

**Comment by kynrai** [comment]
Hello, turns out the problem was when go run main.go is run outside of the directory main is in, for example go run some/dir/main.go --provision will succeed but ultimatly the lambda function will not work. feel free to close this issue. however perhaps the directory issue could be looked at as it does deploy fine but fails when invoked.

**Comment by mweagle** [comment]
Definitely - updated the title and will take a look.  Thanks for filing.

**Comment by mweagle** [comment]
I think I've been able to reproduce this - if the working directory includes a `main`, but `go run` is provided a relative/absolute path to an alternative Sparta source, the resulting binary is malformed. I don't know of a way to get the absolute `go run` input path to set `cwd` during a Sparta build.

Added a [FAQ](http://gosparta.io/docs/faq/#what-working-directory-should-i-use) entry and closing. 

I would like to make this work, so if you know of a way to work around this pease LMK and reopen the issue.  Thanks.
